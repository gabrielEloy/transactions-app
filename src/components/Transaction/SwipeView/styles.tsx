import {StyleSheet} from 'react-native';
import {COLORS} from 'src/themes/main';

const styles = StyleSheet.create({
  container: {
    width: 100,
    justifyContent: 'center',
    backgroundColor: 'red',
    height: 73,
    transform: [{translateX: -10}, {translateY: 6}],
    borderTopRightRadius: 5,
    borderBottomRightRadius: 5,
    alignItems: 'center',
  },
  deleteText: {
    color: COLORS.light,
  },
});

export default styles;
