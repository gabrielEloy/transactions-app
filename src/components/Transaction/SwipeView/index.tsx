import React from 'react';
import {TouchableOpacity, Text} from 'react-native';
import ITransactions from 'interfaces/ITransactions';
import styles from './styles';

interface Props {
  onPress: (transaction: ITransactions) => void;
}

const SwipeView = ({onPress}: Props) => (
  <TouchableOpacity
    onPress={onPress}
    style={styles.container}
    testID="deleteTransaction">
    <Text style={styles.deleteText}>Delete</Text>
  </TouchableOpacity>
);

export default SwipeView;
