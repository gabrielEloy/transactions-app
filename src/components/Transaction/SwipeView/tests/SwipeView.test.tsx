import React from 'react';
import {render} from 'react-native-testing-library';
import SwipeView from '../index';

describe('Button', () => {
  test('Should render correctly', () => {
    const tree = render(<SwipeView onPress={() => console.log('oi')} />);
    expect(tree).toMatchSnapshot();
  });
});
