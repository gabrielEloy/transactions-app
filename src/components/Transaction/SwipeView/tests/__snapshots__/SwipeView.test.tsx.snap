// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Should render correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "red",
      "borderBottomRightRadius": 5,
      "borderTopRightRadius": 5,
      "height": 73,
      "justifyContent": "center",
      "opacity": 1,
      "transform": Array [
        Object {
          "translateX": -10,
        },
        Object {
          "translateY": 6,
        },
      ],
      "width": 100,
    }
  }
  testID="deleteTransaction"
>
  <Text
    style={
      Object {
        "color": "#FBFBFB",
      }
    }
  >
    Delete
  </Text>
</View>
`;
